src/gui/mainwindow.cpp

```diff
@@ -213,8 +225,19 @@ MainWindow::MainWindow(IGUIApplication *app, WindowState initialState)
         UIThemeManager::instance()->getIcon(u"folder-remote"_qs),
 #endif
         tr("Transfers"));
-
-    connect(m_searchFilter, &LineEdit::textChanged, m_transferListWidget, &TransferListWidget::applyNameFilter);
+    // Filter types
+    QVector<TransferListModel::Column> filterTypes = {TransferListModel::Column::TR_NAME, TransferListModel::Column::TR_SAVE_PATH};
+    for (TransferListModel::Column type : filterTypes)
+    {
+        QString typeName = m_transferListWidget->getSourceModel()->headerData(type, Qt::Horizontal, Qt::DisplayRole).value<QString>();
+        m_filterBy->addItem(typeName, type);
+    }
+    auto applyFilter = [this]()
```

Comment:
Please insert blank line above this one.

Reply:
And maybe declare `MainWindow::applyTransferListFilter()` instead? Then you could use it in all the similar cases, e.g.:
```c++
connect(useRegexAct, &QAction::toggled, this, MainWindow::applyTransferListFilter);
```